Applications/Viqui/vqCore.cxx

```diff
@@ -3387,6 +3393,37 @@ void vqCore::generateReport(QString path, bool generateVideo)
   this->exportKml(path);
 }
 
+//-----------------------------------------------------------------------------
+void vqCore::exportSvm(QString path)
+{
+  QFile file(path);
+  if (!file.open(QIODevice::WriteOnly |
+                 QIODevice::Truncate))
+    {
+    qDebug() << file.errorString();
+    QMessageBox::warning(0, QString(), "Unable to write SVM file.");
+    return;
+    }
+
+  if (this->SavedQueryPlan.isSimilarityQuery())
+    {
+    auto& query = *this->SavedQueryPlan.constSimilarityQuery();
+
+    const int iqrModelSize = static_cast<int>(query.IqrModel.size());
+    if (iqrModelSize)
+      {
+      const char* rawModel =
```

Comment:
Please use `std::vector::data` instead. (It's new in C++11, but appears to be supported by GCC 4.8, at least based on [testing on godbolt](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(j:1,lang:c%2B%2B,source:'%23include+%3Cvector%3E%0A%0Avoid+foo()%0A%7B%0A++++std::vector%3Cint%3E+v%3B%0A++++v.data()%3B%0A%7D%0A'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:g482,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',trim:'0'),lang:c%2B%2B,libs:!(),options:'-std%3Dc%2B%2B0x',source:1),l:'5',n:'0',o:'x86-64+gcc+4.8.2+(Editor+%231,+Compiler+%231)+C%2B%2B',t:'0')),k:50,l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compiler:1,editor:1,wrap:'1'),l:'5',n:'0',o:'%231+with+x86-64+gcc+4.8.2',t:'0')),header:(),l:'4',m:50,n:'0',o:'',s:0,t:'0')),k:50,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4).)

This also makes the temporary likely superfluous.